### PR TITLE
Kafka 5854: Handle authentication failures in KafkaAdminClient

### PR DESCRIPTION
**Pull request by rajinisivaram**
This is currently on top of the commits from #3832.

### CODE REVIEW

**Comment by rajinisivaram** [comment]
@vahidhashemian Can you take a look (just the last commit)? We will try and get #3832 merged today and do this tomorrow perhaps.

File: core/src/test/scala/integration/kafka/api/SaslClientsWithInvalidCredentialsTest.scala [hunk 1]
```diff
@@ -95,6 +98,34 @@ class SaslClientsWithInvalidCredentialsTest extends IntegrationTestHarness with
...
+    val props = TestUtils.adminClientSecurityConfigs(securityProtocol, trustStoreFile, clientSaslProperties)
+    props.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
+    val adminClient = AdminClient.create(props)
+
+    def describeTopic(): Unit = {
+      try {
+        val response = adminClient.describeTopics(Collections.singleton(topic)).all.get
+        assertEquals(1, response.size)
+        response.asScala.foreach { case (topic, description) =>
+          assertEquals(numPartitions, description.partitions)
```
Comment (vahidhashemian) [inline core/src/test/scala/integration/kafka/api/SaslClientsWithInvalidCredentialsTest.scala:111]: The types don't seem to match, guess it should be `description.partitions.size`.

**Review by vahidhashemian — COMMENTED** [review]
Thanks for the PR. It looks good to me. There is a minor issue in one of the tests that you probably have fixed already, since I think it's breaking the test.

**Comment by rajinisivaram** [comment]
Closing for now, will rebase on the changes from #3832 and submit when ready.

**Pull request closed without merging**